build_deps/build_pip_pkg.sh

```diff
@@ -0,0 +1,84 @@
+#!/usr/bin/env bash
+# Copyright 2022 The KerasCV Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+set -e
+set -x
+
+PLATFORM="$(uname -s | tr 'A-Z' 'a-z')"
+function is_windows() {
```

Comment:
Some high level question on this:
1) does this build script follow some other scripts? It would be nice to point out the reference so users can follow a pattern
2) do we really want to support windows and mac builds? This will put more burden on maintainence and doesn't give immediate benefit.

Reply:
1. This is adapted from TFA. I'm adding a link to the original script and a brief blurb about what this does.

2. We don't necessarily need to ship binaries for Windows and Mac OS, but including this makes it a lot easier for people to install from source it they want. I'm not sure if it makes sense to offer pre-built wheels for these environments, as I agree that it does add maintenance cost for not a ton of benefit. But it doesn't cost much to support it in the setup script IMO

Reply:
I suppose that these things are ported from Addons: 
https://github.com/tensorflow/addons/blob/master/build_deps/build_pip_pkg.sh

Reply:
The pain is to support those if TF drops pre-built wheels for them